app/presenters/idv/welcome_presenter.rb

```diff
@@ -0,0 +1,81 @@
+module Idv
+  class WelcomePresenter
+    include ActionView::Helpers::TranslationHelper
+    include Rails.application.routes.url_helpers
+    include LinkHelper
+    include ActionView::Helpers::UrlHelper
+
+    attr_accessor :url_options
+
+    def initialize(sp_session)
+      @sp_session = sp_session
+      @url_options = {}
+    end
+
+    def sp_name
+      sp_session.sp_name || APP_NAME
+    end
+
+    def title
+      t('doc_auth.headings.welcome', sp_name: sp_name)
+    end
+
+    def selfie_required?
+      sp_session.selfie_required?
+    end
+
+    def explanation_text(help_link)
+      if selfie_required?
+        t(
+          'doc_auth.info.stepping_up_html',
+          sp_name:,
+          link_html: help_link,
+        )
+      else
+        t(
+          'doc_auth.info.getting_started_html',
+          sp_name: sp_name,
+          link_html: help_link,
+        )
+      end
+    end
+
+    def bullet_header(index)
```

Comment:
I suspect that these could return an array and we could call `each_with_index` in the view.

Reply:
☝️ this is more of a take it or leave it comment. I have a personal preference for `#[]` when dealing with things that have an index.

Reply:
Yeah, that cleaned up nicely.